File: gatekeeper/src/main/java/com/greyblocks/gatekeeper/AppAuthenticatorService.java
package com.greyblocks.gatekeeper;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

/**
 * Created by dyoed on 3/22/17.
 */

public class AppAuthenticatorService extends Service {

    @Override
    public IBinder onBind(Intent intent) {
        return new AccountAuthenticator(this.getApplication()).getIBinder();
    }
}

File: gatekeeper/src/main/java/com/codecodecoffee/gatekeeper/AppAuthenticatorService.kt
package com.codecodecoffee.gatekeeper

import android.app.Service
import android.content.Intent
import android.os.IBinder

/**
 * Created by dyoed on 3/22/17.
 */
class AppAuthenticatorService : Service() {
    override fun onBind(intent: Intent): IBinder {
        return AccountAuthenticator(this.application).iBinder
    }
}